homeassistant/components/elkm1/__init__.py

```diff
@@ -279,8 +279,10 @@ def _keypad_changed(keypad: Element, changeset: dict[str, Any]) -> None:
         keypad.add_callback(_keypad_changed)
 
     try:
-        if not await async_wait_for_elk_to_sync(elk, LOGIN_TIMEOUT, SYNC_TIMEOUT):
-            return False
+        await ElkSyncWaiter(elk, LOGIN_TIMEOUT, SYNC_TIMEOUT).async_wait()
+    except LoginFailed:
+        _LOGGER.error("ElkM1 login failed for %s", conf[CONF_HOST])
+        return False
```

Comment:
in the future we could raise ConfigEntryAuthFailed but this integration doesn't support reauth right now